spec/features/collection_templates/template_index_spec.rb

```diff
@@ -1,36 +1,50 @@
 # The tests in this file verify that the index page contains the expected elements
 
 describe 'When visiting the template index page', js:true do
-  before do
-    login
-    create(:full_collection_template, template_name: 'An Example Template')
-    visit collection_templates_path
-  end
-
   context 'when there is one record' do
+    before do
+      login
+      create(:full_collection_template, template_name: 'An Example Template')
+      visit collection_templates_path
+    end
+
     it 'has expected links for one record' do
       # Create button is actually a link
       expect(page).to have_link('Create a Collection Template', href: '/collection_templates/new')
       expect(page).to have_link('An Example Template', href: '/collection_templates/1')
       expect(page).to have_link('Edit', href: '/collection_templates/1/edit')
       expect(page).to have_link('Delete', href: '#delete-template-modal-0')
     end
-  end
 
-  context 'when there are multiple records' do
-    before do
-      create(:full_collection_template, template_name: 'An Example Template2')
-      visit collection_templates_path
+    it 'the user can access the edit record page' do
+      click_on 'Edit'
+
+      expect(page).to have_content('An Example Template')
+      expect(page).to have_content('Template Name')
     end
 
-    it 'has expected links for both records' do
-      expect(page).to have_link('An Example Template', href: '/collection_templates/1')
-      expect(page).to have_link('Edit', href: '/collection_templates/1/edit')
-      expect(page).to have_link('Delete', href: '#delete-template-modal-0')
+    it 'the user can access the view/show record page' do
+      click_on 'An Example Template'
+
+      expect(page).to have_content('An Example Template') # in the breadcrumbs
+      expect(page).to have_content('Metadata Preview')
+    end
+    
+    context 'when there are multiple records' do
+      before do
+        create(:full_collection_template, template_name: 'An Example Template2')
+        visit collection_templates_path
+      end
+
+      it 'has expected links for both records' do
```

Comment:
The way this is done is fine. Another way (and possibly more brittle) might be https://github.com/nasa/mmt/blob/90df72221f0948fead9df71457abee7deb9c81af/spec/features/groups/filtering_groups_spec.rb#L38.

Reply:
I would definitely be more inclined to do that as the number of things I was testing increased.  Thank you for sharing.